app-admin/logstash-bin/files/logstash-plugin.eselect

```diff
@@ -0,0 +1,77 @@
+# -*-eselect-*-  vim: ft=eselect
+# Copyright 2005-2017 Gentoo Foundation
+# Distributed under the terms of the GNU GPL version 2 or later
+
+DESCRIPTION="Manage user installed plugins of logstash"
+MAINTAINER="hydrapolic@gmail.com"
+
+PLUGINS="/var/lib/logstash/plugins-list.txt"
+LS_INSTALL_DIR="/opt/logstash"
+
+ls_plugin_wrapper() {
+  JARS_SKIP=true "${LS_INSTALL_DIR}/bin/logstash-plugin" "$1" "$2"
+}
+
+### add action
+describe_add() {
+  echo "Add plugin"
+}
+
+describe_add_parameters() {
+  echo "<plugin>"
+}
+
+do_add() {
+  local x
+  for x; do
+    echo "$x" >> ${PLUGINS} || die
+    ls_plugin_wrapper install "$x"
+  done
+
+  sort -u "${PLUGINS}" > "${PLUGINS}.tmp" || die
+  mv "${PLUGINS}.tmp" "${PLUGINS}" || die
+}
+
+### list action
+describe_list() {
+  echo "List user installed plugins"
+}
+
+do_list() {
+  if [[ -f "${PLUGINS}" ]]; then
+    cat "${PLUGINS}"
+  fi
+}
+
+### reinstall action
+describe_reinstall() {
+  echo "Reinstall plugins"
+}
+
+do_reinstall() {
+  if [[ -f "${PLUGINS}" ]]; then
+    local plugin
+    while read plugin; do
+      ls_plugin_wrapper install "${plugin}"
+    done < "${PLUGINS}"
+  fi
+}
+
+### remove action
+describe_remove() {
+  echo "Remove plugin"
+}
+
+describe_remove_parameters() {
+  echo "<plugin>"
+}
+
+do_remove() {
+  if [[ -f "${PLUGINS}" ]]; then
+    local x
+    for x; do
+      sed -i "/^$x\$/d" "${PLUGINS}" || die
```

Comment:
I think it'd be reasonable to emit a warning if you try to remove a plugin that's not there.